File: mobile/android/air/SubModules/WalletBaseContext/src/main/java/org/mytonwallet/app_air/walletbasecontext/models/MBaseCurrency.kt
package org.mytonwallet.app_air.walletbasecontext.models

import org.mytonwallet.app_air.walletbasecontext.localization.LocaleController

enum class MBaseCurrency(val currencyCode: String) {
    USD("USD"),
    EUR("EUR"),
    RUB("RUB"),
    CNY("CNY"),
    BTC("BTC"),
    TON("TON");

    val sign: String
        get() = when (this) {
            USD -> "$"
            EUR -> "€"
            RUB -> "₽"
            CNY -> "¥"
            BTC -> "BTC"
            TON -> "TON"
        }

    val decimalsCount: Int
        get() = when (this) {
            BTC -> 6
            else -> 2
        }

    val currencySymbol: String
        get() = when (this) {
            USD -> LocaleController.getString("USD")
            EUR -> LocaleController.getString("EUR")
            RUB -> LocaleController.getString("RUB")
            CNY -> LocaleController.getString("CNY")
            BTC -> LocaleController.getString("BTC")
            TON -> LocaleController.getString("TON")
        }

    val currencyName: String
        get() = when (this) {
            USD -> LocaleController.getString("US Dollar")
            EUR -> LocaleController.getString("Euro")
            RUB -> LocaleController.getString("Russian Ruble")
            CNY -> LocaleController.getString("Chinese Yuan")
            BTC -> LocaleController.getString("Bitcoin")
            TON -> LocaleController.getString("Toncoin")
        }

    companion object {
        fun parse(value: String) = try {
            valueOf(value)
        } catch (_: Throwable) {
            USD
        }
    }
}

File: mobile/ios/Air/SubModules/WalletCore/Models/MBaseCurrency.swift
//
//  MBaseCurrency.swift
//  WalletCore
//
//  Created by Sina on 3/26/24.
//

import Foundation
import WalletContext

public let DEFAULT_PRICE_CURRENCY = MBaseCurrency.USD

public enum MBaseCurrency: String, Equatable, Hashable, Codable, Sendable, Identifiable, CaseIterable {
    case USD = "USD"
    case EUR = "EUR"
    case RUB = "RUB"
    case CNY = "CNY"
    case BTC = "BTC"
    case TON = "TON"
    
    public var sign: String {
        switch self {
        case .USD:
            return "$"
        case .EUR:
            return "€"
        case .RUB:
            return "₽"
        case .CNY:
            return "¥"
        case .BTC:
            return "BTC"
        case .TON:
            return "TON"
        }
    }
    
    public var decimalsCount: Int {
        switch self {
        case .BTC:
            6
        default:
            2
        }
    }
    
    public var symbol: String {
        return rawValue
    }

    public var name: String {
        switch self {
        case .USD:
            return lang("US Dollar")
        case .EUR:
            return lang("Euro")
        case .RUB:
            return lang("Russian Ruble")
        case .CNY:
            return lang("Chinese Yuan")
        case .BTC:
            return lang("Bitcoin")
        case .TON:
            return lang("Toncoin")
        }
    }
    
    public var id: Self { self }
    
    public var fallbackExchangeRate: Double {
        switch self {
        case .USD:
            1.0
        case .EUR:
            1.0 / 1.1
        case .RUB:
            80.0
        case .CNY:
            7.2
        case .BTC:
            1.0 / 100_000.0
        case .TON:
            1.0 / 3.0
        }
    }
}


